### PR TITLE
Iterator over integer-valued line.

### PR DESCRIPTION
**Pull request by tinevez**
An iterable over a quasi Bresenham line.

It will iterate exactly once over all the integer locations on a line between in proper order from the specified start to the specified end points, included.

This implementation uses floating-point logic instead of the pure integer logic of Bresenham line (Wikipedia) but the results are quasi identical and the performance penalty small.

### CODE REVIEW

File: src/main/java/net/imglib2/roi/geom/integer/Line.java [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+ * <p>
+ * This implementation uses floating-point logic instead of the pure integer
+ * logic of Bresenham line (<a href=
+ * "https://en.wikipedia.org/wiki/Bresenham%27s_line_algorithm">Wikipedia</a>)
+ * but the results are quasi identical and the performance penalty small.
+ * 
+ * @author Jean-Yves Tinevez
+ *
+ */
+public class Line implements IterableInterval< Void >
```
Comment (awalter17) [inline src/main/java/net/imglib2/roi/geom/integer/Line.java]: I wouldn't call this class `Line` since there is already a `Line` in imglib2-roi. Maybe call this `IterableLine` or something else?

Also if you extend `AbstractInterval` you can remove a lot of the `Interval` boilerplate code.

File: src/main/java/net/imglib2/roi/geom/integer/Line.java [hunk 2]
```diff
@@ -0,0 +1,415 @@
...
+		return localizingCursor();
+	}
+
+	@Override
+	public Cursor< Void > localizingCursor()
+	{
+		return new LineCursor();
+	}
+
+	private final class LineCursor implements Cursor< Void >
```
Comment (awalter17) [inline src/main/java/net/imglib2/roi/geom/integer/Line.java]: You could extend `AbstractCursor`, but I'm not sure if it will simplify this much since you need to do the rounding.

File: src/test/java/net/imglib2/roi/geom/integer/LineTest.java [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+import net.imglib2.img.ImgFactory;
+import net.imglib2.img.array.ArrayImg;
+import net.imglib2.img.array.ArrayImgFactory;
+import net.imglib2.img.array.ArrayImgs;
+import net.imglib2.img.basictypeaccess.array.ByteArray;
+import net.imglib2.roi.Regions;
+import net.imglib2.type.numeric.integer.UnsignedByteType;
+import net.imglib2.util.Util;
+
+public class LineTest
```
Comment (awalter17) [inline src/test/java/net/imglib2/roi/geom/integer/LineTest.java]: It would be nice to also test vertical and horizontal lines. It looks like your code should cover these edge cases, but it would be good to formally test them as well.

**Comment by tinevez** [comment]
Done!

**Comment by ctrueden** [comment]
@tpietzsch @awalter17 Any reason this cannot be merged now?

**Comment by stelfrich** [comment]
This would be so nice to have! Just the other week I had hack together a Bresenham implementation..

**Comment by tpietzsch** [comment]
Thank you @tinevez!
(And thank you @stelfrich for bumping this again. Sorry for taking so long.)
